sdks/python/apache_beam/transforms/core.py

```diff
@@ -1358,6 +1358,11 @@ def expand(self, pcoll):
         *args, **kwargs)
 
   def to_runner_api_parameter(self, context):
+    if self.side_inputs:
```

Comment:
This looks right to me. I have not realized before that using side inputs will prevent lifting but it makes sense.

However, I do not think this is clean. Because the next if self.args or self.kwargs is also practically checking for the same thing. args and kwargs are used for passing side inputs and do not have another purpose (https://github.com/apache/beam/blob/e69a353c94d0a54d3f7194736433e4dbac87303f/sdks/python/apache_beam/transforms/core.py#L1251)

You probably need to edit that if instead.

Reply:
I actually tried that already, but I believe some other case triggers that statement even when there are no side inputs, because this would cause errors with some tests I tried it on that were avoided with this version. I believe something other than side inputs can be present in the args, or they may be modified at some point, but I didn't have enough context to track down the specifics. Also it didn't seem worth the effort just to make the code slightly more easy to read.

I noticed @robertwb comes back tomorrow so he might be able to explain why that happens. If not I would prefer to just leave a comment about it and leave the code as-is.

Reply:
On second thought, I just noticed that the PR for combiner lifting went through. In order to avoid issues with pipelines with side inputs (combiner lifting breaks when a combine has side inputs), I'd like to get this in sooner, so I think I'd rather leave a comment and ask Robert later.